pkgs/by-name/rp/rpm-sequoia/package.nix

```diff
@@ -0,0 +1,87 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+  darwin,
+  nettle,
+  nix-update-script,
+  rustPlatform,
+  pkg-config,
+  runCommand,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rpm-sequoia";
+  version = "1.7.0";
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = "rpm-sequoia";
+    rev = "v${version}";
+    hash = "sha256-AZCsboUv4muKOw5El2Hw5O1cvAgD3JhBppacrQCJT2k=";
+  };
+
+  cargoHash = "sha256-0yO1+OAkXje/ir8i8URVhIcW8gwXleYx+XL1U4bjtXk=";
+
+  nativeBuildInputs =
+    [
+      pkg-config
+      rustPlatform.bindgenHook
+    ]
+    ++ lib.optionals
+      (stdenv.hostPlatform != stdenv.buildPlatform && stdenv.buildPlatform.canExecute stdenv.hostPlatform)
+      [
+        # Tests will try to shell out to objdump to check the exported symbols
+        # fixup pkgsLLVM.rpm-sequoia
+        (runCommand "objdump" { } ''
+          mkdir -p $out/bin
+          ln -s ${stdenv.cc.bintools}/bin/${stdenv.cc.bintools.targetPrefix}objdump $out/bin/objdump
+        '')
+      ];
+
+  propagatedBuildInputs = [ nettle ];
+
+  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin (
+    with darwin.apple_sdk.frameworks;
+    [
+      Security
+      SystemConfiguration
+    ]
+  );
+
+  # Tests will parse the symbols, but the library name is hardcoded to librpm_sequoia.so
+  # which is invalid on darwin.
+  doCheck = !stdenv.hostPlatform.isDarwin;
+
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  # Ensure the generated .pc file gets the correct prefix
+  env.PREFIX = placeholder "out";
+
+  preInstall =
+    # Install the generated pc file for consumption by the dependents
+    ''
+      install -Dm644 target/release/rpm-sequoia.pc -t $dev/lib/pkgconfig
+    ''
+    +
+      # Dependents will rely on the versioned symlinks
+      ''
+        install -d $out/lib
+        find target/release/ \
+          -maxdepth 1 \
+          -type l -name 'librpm_sequoia.*' \
+          -exec cp --no-dereference {} $out/lib/ \;
+      '';
```

Comment:
Do we know how this works on e.g. Fedora? Is the main installation procedure not meant to handle this?

Reply:
https://src.fedoraproject.org/rpms/rust-rpm-sequoia/blob/f41/f/rust-rpm-sequoia.spec#_81

This is done manually there too.

Reply:
Added the link to fedora in a comment